In [1]:
import json

In [5]:
with open('llama_preliminary_1.json', 'r') as f:
    data1 = json.load(f)


len(data1)

2090

In [11]:
with open('llama_preliminary_2.json', 'r') as f:
    data2 = json.load(f)


total = len(data2)
total

2090

In [14]:
chars = []
words = []
distribution1 = {}
for element in data1:
    sentence = element['base_sentence']
    num_chars = len(sentence)
    num_words = len(sentence.split())

    chars.append(num_chars)
    words.append(num_words)

    # distribution
    label = element['likelihoods']['predicted_label']
    if label in distribution1:
        distribution1[label] += 1
    else:
        distribution1[label] = 1

print(f'There are on average {sum(chars)/len(chars):.2f} characters per example')
print(f'There are on average {sum(words)/len(words):.2f} words per example')

There are on average 56.75 characters per example
There are on average 9.16 words per example


In [15]:
chars = []
words = []
distribution2 = {}
for element in data2:
    sentence = element['base_sentence']
    num_chars = len(sentence)
    num_words = len(sentence.split())

    chars.append(num_chars)
    words.append(num_words)

    #distribution
    label = element['likelihoods']['predicted_label']
    if label in distribution2:
        distribution2[label] += 1
    else:
        distribution2[label] = 1

print(f'There are on average {sum(chars)/len(chars):.2f} characters per example')
print(f'There are on average {sum(words)/len(words):.2f} words per example')

There are on average 64.29 characters per example
There are on average 10.40 words per example


In [16]:
distribution1

{'stereotype': 1260, 'anti-stereotype': 733, 'unrelated': 97}

In [17]:
distribution2

{'stereotype': 1254, 'anti-stereotype': 731, 'unrelated': 105}

- There are 2090 examples for each dataset, which are all used in the testing set. This is because we are evaluating the bias on pretrained llms. 

For data1:
- There are on average 56.75 characters per example
- There are on average 9.16 words per example
- The predicted distribution is {'stereotype': 1260, 'anti-stereotype': 733, 'unrelated': 97}

For data2:
- There are on average 64.29 characters per example
- There are on average 10.40 words per example
- The predicted distribution is {'stereotype': 1254, 'anti-stereotype': 731, 'unrelated': 105}

StereoSet is the original dataset. The second dataset is generated using GPT-4. 